[2024-04-16T09:41:57.511+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-16T09:41:57.528+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:28:00+00:00 [queued]>
[2024-04-16T09:41:57.536+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:28:00+00:00 [queued]>
[2024-04-16T09:41:57.536+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-16T09:41:57.548+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): hello2> on 2024-04-16 06:28:00+00:00
[2024-04-16T09:41:57.555+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=3525) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-04-16T09:41:57.557+0000] {standard_task_runner.py:63} INFO - Started process 3536 to run task
[2024-04-16T09:41:57.556+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spotify_final_dag', 'hello2', 'scheduled__2024-04-16T06:28:00+00:00', '--job-id', '894', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmpdv_xl_5i']
[2024-04-16T09:41:57.557+0000] {standard_task_runner.py:91} INFO - Job 894: Subtask hello2
[2024-04-16T09:41:57.593+0000] {task_command.py:426} INFO - Running <TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:28:00+00:00 [running]> on host 52651c1f9987
[2024-04-16T09:41:57.679+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spotify_final_dag' AIRFLOW_CTX_TASK_ID='hello2' AIRFLOW_CTX_EXECUTION_DATE='2024-04-16T06:28:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-04-16T06:28:00+00:00'
[2024-04-16T09:41:57.680+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-16T09:42:01.227+0000] {logging_mixin.py:188} INFO - There was a problem with the connection. Trial number 1, Status Code: 401
[2024-04-16T09:42:04.570+0000] {logging_mixin.py:188} INFO - There was a problem with the connection. Trial number 2, Status Code: 401
[2024-04-16T09:42:04.572+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-16T09:42:04.573+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/get_data.py", line 98, in processes
    album_data = get_album(connection())
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/get_data.py", line 11, in get_album
    for album in data["items"]:
                 ~~~~^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
[2024-04-16T09:42:04.582+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=spotify_final_dag, task_id=hello2, execution_date=20240416T062800, start_date=20240416T094157, end_date=20240416T094204
[2024-04-16T09:42:04.592+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 894 for task hello2 ('NoneType' object is not subscriptable; 3536)
[2024-04-16T09:42:04.629+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-04-16T09:42:04.641+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-16T09:42:04.643+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-04-16T09:48:49.270+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-16T09:48:49.296+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:28:00+00:00 [queued]>
[2024-04-16T09:48:49.312+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:28:00+00:00 [queued]>
[2024-04-16T09:48:49.312+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-16T09:48:49.330+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): hello2> on 2024-04-16 06:28:00+00:00
[2024-04-16T09:48:49.341+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=4977) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-04-16T09:48:49.342+0000] {standard_task_runner.py:63} INFO - Started process 5004 to run task
[2024-04-16T09:48:49.343+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spotify_final_dag', 'hello2', 'scheduled__2024-04-16T06:28:00+00:00', '--job-id', '1348', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmponj67rxk']
[2024-04-16T09:48:49.344+0000] {standard_task_runner.py:91} INFO - Job 1348: Subtask hello2
[2024-04-16T09:48:49.399+0000] {task_command.py:426} INFO - Running <TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:28:00+00:00 [running]> on host 52651c1f9987
[2024-04-16T09:48:49.537+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spotify_final_dag' AIRFLOW_CTX_TASK_ID='hello2' AIRFLOW_CTX_EXECUTION_DATE='2024-04-16T06:28:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-04-16T06:28:00+00:00'
[2024-04-16T09:48:49.539+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-16T09:48:49.980+0000] {logging_mixin.py:188} INFO - Connection successful
[2024-04-16T09:48:50.458+0000] {logging_mixin.py:188} INFO - Connection successful
[2024-04-16T09:48:50.886+0000] {logging_mixin.py:188} INFO - Connection successful
[2024-04-16T09:48:50.891+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.893+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.904+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.904+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.906+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.907+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.908+0000] {python.py:237} INFO - Done. Returned value was: None
[2024-04-16T09:48:50.908+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-16T09:48:50.915+0000] {taskinstance.py:1205} INFO - Marking task as SUCCESS. dag_id=spotify_final_dag, task_id=hello2, execution_date=20240416T062800, start_date=20240416T094849, end_date=20240416T094850
[2024-04-16T09:48:50.962+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2024-04-16T09:48:50.974+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-16T09:48:50.976+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
